src/lib/swapper/swappers/CowSwapper/utils/helpers/helpers.ts

```diff
@@ -132,13 +128,10 @@ export const assertValidTrade = ({
 }: {
   buyAsset: Asset
   sellAsset: Asset
-  supportedChainIds: CowChainId[]
+  supportedChainIds: ChainId[]
```

Comment:
Can't we keep the previous narrowed down type here?

Reply:
not without casting somewhere upstream which negates its usefulness. This function enforces the correct types, but ideally we'd use a type assertion such that it allows the compiler to know the arguments are of the type we require, which isnt compatible with the monadic error paradigm we've gone with (since a type assertion throws errors rather than returns Err)